telemetry-impls/summarize/action.yml

```diff
@@ -1,149 +1,37 @@
 name: 'Telemetry summarize'
 description: |
   Consumes job info, parses into spans, and pushes spans.
-inputs:
-  cert_concat:
-    description: Concatenation of certs (CA;Client;ClientKey)
-  extra_attributes:
-    description: |
-      Additional attributes to add to OTEL_RESOURCE_ATTRIBUTES.
-      See https://opentelemetry.io/docs/languages/sdk-configuration/general/#otel_resource_attributes
-
-      Do not include any leading or trailing join characters (,)
 
 runs:
   using: 'composite'
   steps:
-    - uses: ./shared-actions/telemetry-impls/ensure-otel-cli-available
+    - name: Setup python
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.11
+        cache: 'pip'
+        cache-dependency-path: './shared-actions/telemetry-impls/summarize/requirements.txt'
+    - name: Install dependencies
+      run: pip install -r './shared-actions/telemetry-impls/summarize/requirements.txt'
+      shell: bash
     # Writes JSON file that jobs below consume
     - uses: ./shared-actions/telemetry-impls/github-actions-job-info
       id: github-job-info
-
-    - name: Add attribute metadata beyond the stashed basic stuff
-      shell: bash
-      run:
-          attributes="${OTEL_RESOURCE_ATTRIBUTES}";
-          labels="$(jq -r '.labels | join(" ")' job_info.json)";
-          if [ "${labels}" != "" ]; then
-            attributes="${attributes},rapids.labels=${labels}";
-          fi;
-          if [ "${{ inputs.extra_attributes }}" != "" ]; then
-            attributes="${attributes},${{ inputs.extra_attributes }}";
-          fi;
-          echo OTEL_RESOURCE_ATTRIBUTES="$(echo "${attributes}" | sed 's/^,//')" >> ${GITHUB_ENV};
-
-    - uses: ./shared-actions/telemetry-impls/set-otel-service-name
-      if: env.OTEL_SERVICE_NAME == ''
-
-    # OpenTelemetry implementations look for these variable names to be set to the paths
-    #    of the files with the mTLS certs and keys. We base64 encode them when storing them
-    #    to avoid issues with quoting and multiline text.
-    # If these env vars are not set, then otel-cli will not attempt to use mTLS.
-    - name: Write certificate files for mTLS
-      if: "${{ inputs.cert_concat }} != ''"
+      with:
+        all_jobs: true
+    - name: Upload job JSON file if debugging
+      uses: actions/upload-artifact@v4
+      if: runner.debug == '1'
+      with:
+        name: telemetry-tools-all_jobs.json
+        path: all_jobs.json
+    # This downloads ALL of the files that we have collected from each job.
+    - uses: actions/download-artifact@v4
```

Comment:
Given that we intend on eventually switching to using GitHub artifacts (see https://github.com/rapidsai/ops/issues/2982), I think we should create a prefix specifically for telemetry files and then use this action's `pattern` input to ensure that we don't needlessly download unrelated artifacts.

See here for more details: https://github.com/actions/download-artifact?tab=readme-ov-file#download-multiple-filtered-artifacts-to-the-same-directory